File: 2.go
package main

import "fmt"

func main() {
    fmt.Println(fibsum(4000000));
}

func fibsum(limit int) (sum int) {
    sum = 2

    prev := 1
    curr := 2
    temp := 0

    for run := true; run; run = curr < limit {
        temp = prev
        prev = curr
        curr += temp

        if curr % 2 == 0 {
            sum += curr
        }
    }

    return
}

File: 2.php
<?php

$limit = 4000000;
$prev = 1;
$curr = 2;

$sum = 2;

while (true) {
    $tmp = $prev;
    $prev = $curr;
    $curr = $prev + $tmp;

    if ($curr > $limit) break;

    if ($curr % 2 == 0) $sum += $curr;
}

var_dump($sum); die;
